R/ancombc_prep.R

```diff
@@ -30,7 +30,7 @@ tse_construct = function(data, assay_name, tax_level, phyloseq) {
 
         # Check if agglomeration should be performed
         if (is.null(tax_level)) {
-            tax_levels = mia::taxonomyRanks(tse)
+            tax_levels = union(mia::taxonomyRanks(tse), colnames(SummarizedExperiment::rowData(tse)))
```

Comment:
Is this needed? the rowData may contain also other features than taxonomic ranks, do we need to define them as tax_levels? What is the intended purpose of this line?

Reply:
Not necessarily